src/DSi.cpp

```diff
@@ -918,41 +1024,81 @@ void ARM9Write8(u32 addr, u8 val)
     switch (addr & 0xFF000000)
     {
     case 0x03000000:
-        if (addr >= NWRAMStart[0][0] && addr < NWRAMEnd[0][0])
+    case 0x03800000:
+        if (SCFG_EXT[0] & (1 << 25))
         {
-            u8* ptr = NWRAMMap_A[0][(addr >> 16) & NWRAMMask[0][0]];
-            if (ptr)
+            if (addr >= NWRAMStart[0][0] && addr < NWRAMEnd[0][0])
             {
-                *(u8*)&ptr[addr & 0xFFFF] = val;
+                // Write to a bank is special, as it writes to all
+                // parts that are mapped and not just the highest priority
+                // See http://melonds.kuribo64.net/board/thread.php?pid=3974#3974
+                // so we need to iterate through all parts and write to all mapped here
+                u32 destPart = (addr >> 15) & NWRAMMask[0][0];
+                for (int page = 0; page < 4; page++)
```

Comment:
this seems to bit intense for every memory write. Would it be possible to do precalculate some of this similar to the way it's [done](https://github.com/Arisotura/melonDS/blob/master/src/GPU.cpp#L530) for [VRAM](https://github.com/Arisotura/melonDS/blob/master/src/GPU.h#L346-L382)?

Reply:
You would need to precalculate a mask for every 32kB within the image size and still would require to do a check for each page. The compare of a single bit (precalculated mappings) or by the 3 bits (mapping bits in page) should be equal in performance.

Reply:
ok well the most common case where there are no weird overlappings is going to be handled fastmem, so atleast with JIT+fastmem it won't make a difference.

Reply:
I reduced the number of comparison (included the disable/assigned core) into the same compare. Thanks for letting me look over it again :-)